dotCMS/src/main/java/com/dotcms/vanityurl/cache/VanityUrlCacheImpl.java

```diff
@@ -0,0 +1,156 @@
+package com.dotcms.vanityurl.cache;
+
+import java.util.List;
+import java.util.Optional;
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+
+/**
+ * This class implements {@link VanityUrlCache} the cache for Vanity URLs. Is used to map the Vanity
+ * URLs path to the Vanity URL content
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 24, 2017
+ */
+public class VanityUrlCacheImpl extends VanityUrlCache {
+
+    private DotCacheAdministrator cache;
+
+    private static final String VANITY_URL_SITE_GROUP = "VanityURLSiteCache";
+    private static final String VANITY_URL_DIRECT_GROUP = "VanityURLDirectCache";
+
+
+
+    public VanityUrlCacheImpl() {
+        cache = CacheLocator.getCacheAdministrator();
+    }
+
+    @Override
+    public String getPrimaryGroup() {
+
+        return VANITY_URL_SITE_GROUP;
+    }
+
+    @Override
+    public String[] getGroups() {
+
+        return new String[] {VANITY_URL_SITE_GROUP, VANITY_URL_DIRECT_GROUP};
+    }
+
+    @Override
+    public void clearCache() {
+        cache.flushGroup(VANITY_URL_SITE_GROUP);
+        cache.flushGroup(VANITY_URL_DIRECT_GROUP);
+
+    }
+
+    @Override
+    public void remove(final Contentlet vanityURL) {
+        if (vanityURL == null || !vanityURL.isVanityUrl()) {
+            return;
+        }
+        Host host = Try.of(() -> APILocator.getHostAPI().find(vanityURL.getHost(), APILocator.systemUser(), false)).getOrNull();
+        Language lang = Try.of(() -> APILocator.getLanguageAPI().getLanguage(vanityURL.getLanguageId())).getOrNull();
+        if (host == null || lang == null) {
+            return;
+        }
+        remove(host, lang);
+
+
+    }
+
+
+    @Override
+    public void remove(final Host vanityHost) {
+        if (vanityHost == null) {
+            return;
+        }
+
+        final List<Language> langs = APILocator.getLanguageAPI().getLanguages();
+
+        for (Language lang : langs) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'lang' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)